payjoin/src/psbt/merge.rs

```diff
@@ -0,0 +1,268 @@
+use std::fmt;
+
+use bitcoin::{OutPoint, Psbt};
+
+#[allow(dead_code)]
+/// Error type for merging two unique unsigned PSBTs
+#[derive(Debug, PartialEq)]
+pub(crate) enum MergePsbtError {
+    /// Input from other PSBT already exists in this PSBT
+    InputAlreadyExists(OutPoint),
+    /// Input is already signed
+    MyInputIsSigned(OutPoint),
+    /// Other PSBT's input is already signed
+    OtherInputIsSigned(OutPoint),
+}
+
+impl fmt::Display for MergePsbtError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::InputAlreadyExists(outpoint) =>
+                write!(f, "input already exists with outpoint: {}", outpoint),
+            Self::MyInputIsSigned(outpoint) =>
+                write!(f, "my input is already signed with outpoint: {}", outpoint),
+            Self::OtherInputIsSigned(outpoint) =>
+                write!(f, "other input is already signed with outpoint: {}", outpoint),
```

Comment:
@spacebear21 I know we've been using e.g. `use MergePsbtError::*` ... `Variant(e) => _` but I don't hate simplicity of `Self::`. It's just two lines saved though and still a lot of repetition. Would you prefer to have the import or `Self` be the consistent style we use across the codebase?